spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/CreateChangelogViewProcedure.java

```diff
@@ -228,22 +247,41 @@ private Dataset<Row> applyChangelogIterator(Dataset<Row> df, Column[] repartitio
             RowEncoder.apply(schema));
   }
 
-  private Dataset<Row> applyCarryoverRemoveIterator(Dataset<Row> df, Column[] repartitionSpec) {
-    Column[] sortSpec = sortSpec(df, repartitionSpec);
+  private Dataset<Row> applyCarryoverRemoveIterator(
+      Dataset<Row> df, Column[] repartitionSpec, boolean netChanges) {
+    Column[] sortSpec = sortSpec(df, repartitionSpec, netChanges);
     StructType schema = df.schema();
 
     return df.repartition(repartitionSpec)
         .sortWithinPartitions(sortSpec)
         .mapPartitions(
             (MapPartitionsFunction<Row, Row>)
-                rowIterator -> ChangelogIterator.removeCarryovers(rowIterator, schema),
+                rowIterator ->
+                    netChanges
+                        ? ChangelogIterator.removeNetCarryovers(rowIterator, schema)
+                        : ChangelogIterator.removeCarryovers(rowIterator, schema),
             RowEncoder.apply(schema));
   }
 
-  private static Column[] sortSpec(Dataset<Row> df, Column[] repartitionSpec) {
-    Column[] sortSpec = new Column[repartitionSpec.length + 1];
-    System.arraycopy(repartitionSpec, 0, sortSpec, 0, repartitionSpec.length);
+  private static Column[] sortSpec(Dataset<Row> df, Column[] repartitionSpec, boolean netChanges) {
+    Column[] sortSpec;
+
+    if (netChanges) {
+      Column changeOrdinal = df.col(MetadataColumns.CHANGE_ORDINAL.name());
+      Preconditions.checkState(
```

Comment:
Is this necessary?  Just asking as we dont do it for CHANGE_TYPE

Reply:
I added it since this method take a flag `netchange`. When the flag is `false`, repartition spec will have `changeOrdinal`. But I'm OK to remove it if you think is not necessary.

Reply:
Removed it anyway.